src/main/java/com/adobe/target/edge/client/http/DefaultTargetHttpClient.java

```diff
@@ -75,15 +80,62 @@ public void addDefaultHeader(String key, String value) {
   }
 
   @Override
-  public <T, R> HttpResponse<R> execute(
-      Map<String, Object> queryParams, String url, T request, Class<R> response) {
-    return unirestInstance.post(url).queryString(queryParams).body(request).asObject(response);
+  public <T, R> ResponseWrapper<R> execute(
+      Map<String, Object> queryParams, String url, T request, Class<R> responseClass) {
+    ResponseWrapper<R> responseWrapper = new ResponseWrapper<>();
+
+    HttpResponse<Object> httpResponse =
+        unirestInstance
+            .post(url)
+            .queryString(queryParams)
+            .body(request)
+            .asObject(
+                rawResponse -> {
+                  TimingTool timer = new TimingTool();
+                  timer.timeStart(TIMING_EXECUTE_REQUEST);
+                  String rawResponseContent = rawResponse.getContentAsString();
+                  R responseBody = getObjectMapper().readValue(rawResponseContent, responseClass);
+                  responseWrapper.setParsingTime(timer.timeEnd(TIMING_EXECUTE_REQUEST));
+                  responseWrapper.setResponseSize(rawResponseContent.length());
```

Comment:
Are you able to use `rawResponse.getContentAsBytes().length` here, but still pass `rawResponse.getContentAsString()` to the ObjectMapper?  Ideally we would get the length of the byte array because I'm not sure if the length will always be the same as the string.

Reply:
When I looked into the library implemenataion, I found following:
rawResponse.getContentAsString() uses stream. The stream gets closed after rawResponse.getContentAsString() is called. That's why we can't use stream again in 
rawResponse.getContentAsBytes() to get its length. Even rawResponse.getContentAsString() calls rawResponse.getContentAsBytes() internally.

To fix the issue, I have  caputured byte[] (rawResponse.getContentAsBytes) and convert byte[] to String to supply it to Object Mapper. I need to identify charset to convert byte[] to String. The charSet extraction was done by library. But getCharSet is marked as protected and not exposed by the library. That's the reason I needed do rewrite some library implementation to getCharSet from rawResponse.